### ISSUE TITLE
Cannot execute action on self-hosted windows x64 machine

### ISSUE THREAD

**Issue by jjhartmann** [game-ci/unity-test-runner#129]
I keep getting this error when running the test-runner on a self-hosted windows x64 machine,

Result:
```
Run game-ci/unity-test-runner@v2
  with:
    githubToken: ***
    customParameters: -testPlatform StandaloneWindows64
    unityVersion: auto
    testMode: all
    artifactsPath: artifacts
    useNetworkHost: false
    checkName: Test Results
  env:
    UNITY_EMAIL: ***
    UNITY_PASSWORD: ***
    UNITY_SERIAL: ***
Error: Currently win32-platform is not supported
```

I tested a custom `cmd` using the Unity CLI, and it executes as expected:
```
- name: Run Unit Tests
        shell: cmd
         run: |
           Unity.exe -runTests -batchmode -projectPath %cd% -testResults %cd%\..\Results\EditModeTests.xml
```

Any insight into the issue?

**Comment by webbertakken** [comment]
We never added support or tested for the javascript part to run on a Windows machine.

Feel free to fork test runner and remove the check that throws this error and check if it works.

**Comment by jongfeelkim-VIRNECT** [comment]
Any support and test plans on Windows?

**Comment by davidmfinol** [comment]
Windows support is now available on `unity-builder`, and some of the required support has been ported over to `unity-test-runner`.

However, there is still a check that enforces only Linux to be used.

Still open to any PR that wants to try removing that check and validating that it works.

**Comment by maishiroma** [comment]
Hey, all! I just wanted to forefront this by saying thanks for creating these GitHub actions for anyone to use for their needs; they've been very helpful in my current work to help simplify some of the steps needed to run Unity in GitHub Actions. 😁 

I stumbled across this issue when I also noticed this action isn't supported by Windows; I'd be interested to see if I can take a look at getting this action supported for windows. 😃 I'll be sure to update this issue once I believe I made some progress on my side.

**Comment by maishiroma** [comment]
Ok, I spent a few days working on the Windows support and I believe I have something that is ready for a review! There's a small nuance I also noticed during the execution of the action, and I went ahead and got them patched out.
- If the following value was in the test results (`<stack-trace><![CDATA[]]></stack-trace>`), the action will completely error out; I'm not sure why this occurs, but a small null checker [here](https://github.com/game-ci/unity-test-runner/compare/main...maishiroma:win_support#diff-956ae71c9a3d1d3bee67f3bf681723cfb71a43bb81939279b199925306955d9d), resolves it.

I'll post up a pull request to get my new changes up. Let me know if there's anything I can clarify on!
